src/main/java/sh/sinux/musicmanager/LibraryStorage/LibraryStorageLinkedList.java

```diff
@@ -80,5 +115,69 @@ public Song[] searchByArtist(String artist) {
         }
         return matchingSongs.toArray(new Song[matchingSongs.size()]);
     }
+    /**
+     * Sorts the songs using a custom comparator (using selection sort)
+     *
+     * @param comparator The comparator to use for sorting
+     */
+    public void selectionSort(Comparator<Song> comparator) {
+        int size = songs.size();
+        // Iterate through each element of the list
+        for (int i = 0; i < size - 1; i++) {
+            // Find the smallest element in the unsorted portion of the list
+            int minIndex = i;
+            for (int j = i + 1; j < size; j++) {
+                Song currentSong = songs.get(j);
+                //smallestSong which has the smallest Unicode value
+                Song smallestSong = songs.get(minIndex);
+                //using the comparator to compare the current song with the smallest song
+                if (comparator.compare(songs.get(j), songs.get(minIndex)) < 0) {
+                    minIndex = j;
+                }
+            }
+            // swap the smallest element with the first element of the unsorted portion of the list ( smallest to the front)
+            if (minIndex != i) {
+                Song temp = songs.get(i);
+                songs.set(i, songs.get(minIndex));
+                songs.set(minIndex, temp);
+            }
+        }
+    }
+
+    public static void main(String[] args) {
+        LibraryStorageLinkedList library = new LibraryStorageLinkedList();
+
+        // Add some songs to the library
+        library.add(new Song("Song A", "Album 2", "Artist 1", "Pop"));
+        library.add(new Song("Song B", "Album 1", "Artist 1", "Pop"));
+        library.add(new Song("Song D", "Album 1", "Artist 3", "Jazz1"));
+        library.add(new Song("Song D", "Album 1", "Artist 3", "Jazz1"));
+        library.add(new Song("Song c", "Album 3", "Artist 1", "Jazz1"));
+        library.add(new Song("Song D", "Album 3", "Artist 1", "Jazz4"));
+
+        // Display unsorted library
+        System.out.println("Unsorted library:");
+        displayLibrary(library);
+
+        // Sort the library using the AlbumComparator
+        Comparator<Song> albumComparator = new AlbumComparator();
+        Comparator<Song> ratingComparator = new RatingComparator();
+        Comparator<Song> genre = new GenreComparator();
+
+        library.selectionSort(albumComparator);
+        System.out.println("Sorted library:(album)");
+        displayLibrary(library);
+        library.selectionSort(genre);
+        System.out.println("Sorted library:(genre)");
+        displayLibrary(library);
+
+    }
+
+    private static void displayLibrary(LibraryStorageLinkedList library) {
```

Comment:
If we do that, we'll have to add it to LibraryStorage interface, and implement in the 2 other classes